File: src/main/java/akostenko/aicars/race/cartest/AccelerationTest.java
package akostenko.aicars.race.cartest;

import akostenko.aicars.race.Driver;

public class AccelerationTest extends Driver {

    private final double targetSpeed; // km/h
    private double time = 0;
    private boolean targetReached;

    public AccelerationTest(double targetSpeedKMH) {
        this.targetSpeed = targetSpeedKMH;
    }

    @Override
    public String getName() {
        return String.format("0-%d kph", (int) targetSpeed) + (targetReached ? String.format(" %.3fs", time) : "");
    }

    @Override
    public double accelerating() {
        return !targetReached ? 1 : 0;
    }

    @Override
    public double breaking() {
        return targetReached ? 1 : 0;
    }

    @Override
    public double steering() {
        return 0;
    }

    @Override
    public void update(double dTime) {
        if (!targetReached) {
            targetReached = getCar().speed().module() * 3.6 >= targetSpeed;
            time += dTime;
        }
    }
}

File: game-engine/src/main/kotlin/akostenko/aicars/race/cartest/AccelerationTest.kt
package akostenko.aicars.race.cartest

import akostenko.aicars.race.Driver

class AccelerationTest(private val targetSpeed: Double // km/h
) : Driver() {
    private var time = 0.0
    private var targetReached: Boolean = false

    override val name: String
        get() = String.format("0-%d kph", targetSpeed.toInt()) + if (targetReached) String.format(" %.3fs", time) else ""

    override fun accelerating(): Double {
        return (if (!targetReached) 1 else 0).toDouble()
    }

    override fun breaking(): Double {
        return (if (targetReached) 1 else 0).toDouble()
    }

    override fun steering(): Double {
        return 0.0
    }

    override fun update(dTime: Double) {
        if (!targetReached) {
            targetReached = car.speed.module() * 3.6 >= targetSpeed
            time += dTime
        }
    }
}
